lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,24 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const prerelease = semver.prerelease(currentValue);
+    if (!prerelease) {
+      newPomVersion = semver.inc(currentValue, bumpVersion, 'SNAPSHOT', false);
```

Comment:
I still don't get this condition. 🤔

in which case we get here, and why those extra arguments?

Reply:
We get here if the version is a release version (for example `1.2.3`).

The extra arguments are used if the `bumpVersion` is `pre*`, in which case it ensures that the prerelease suffix is `-SNAPSHOT` and the `false` value makes sure that no additional counter is added (it would add `-SNAPSHOT.0` if not given).

Reply:
See https://github.com/renovatebot/renovate/pull/27274#issuecomment-1975197843